tests/test_compat.py

```diff
@@ -36,7 +36,7 @@ def itervalues(self):
         return iter(self.values())
 
 
-class TestCaseBase(unittest.TestCase):
+class BaseTestCase:
```

Comment:
I would not rename the name of the class. There is no reason for that.

This class still should be child of `unittest.TestCase`. It is strange that this class is currently not child of `unittest.TestCase`, but it uses many methods from this class. e.g. `self.assertTrue()`

Maybe it will not be possible to use automatic discovery of two test cases (child classes of `TestCaseBase`)

Reply:
If it starts with Test* it will be collected, and if this runs, it causes failures because `config_class` is not set.

Reply:
The class should be subclass of `unittest.TestCase`, then it would be collected anyway. Maybe, if `config_class` is set to `None`, then it could be used for condition for skipping tests. It could be possible to use `unittest.skipIf`.